src/backends/mirror.ts

```diff
@@ -68,8 +70,27 @@ export async function prepareMirrorRequests(
 ) {
   logger.info("[S3 backend] mirroring requests...");
 
-  const backupBuckets = bucketConfig.backups!;
-  for (const backupConfig of backupBuckets) {
+  const replicaBuckets: ReplicaConfig[] = [];
+  for (const replica of globalConfig.replicas) {
+    let primaryBucket = "";
+    if (bucketConfig.typ === "S3BucketConfig") {
+      primaryBucket = bucketConfig.config.bucket;
+    } else {
+      primaryBucket = bucketConfig.config.container;
+    }
+
+    let replicaBucket = "";
+    if (replica.typ === "ReplicaS3Config") {
```

Comment:
prefer a const+ternary if you have 2 values possibles or const+dict lookup if you have more

Reply:
wow, how did I miss that